packages/mattermost-redux/src/utils/post_utils.ts

```diff
@@ -140,16 +140,22 @@ function isJoinLeavePostForUsername(post: Post, currentUsername: string): boolea
         post.props.removedUsername === currentUsername;
 }
 
+export function isPostUploadingFile(post: Post): boolean {
+    return post.id === post.pending_post_id && post.file_client_ids !== undefined && post.file_client_ids.length !== 0;
```

Comment:
better check for `> 0` since any value other than `0` would return true (even `undefined`)
```suggestion
    return post.id === post.pending_post_id && post.file_client_ids?.length > 0;
```

A short explanation why I am not suggesting `return post.id === post.pending_post_id && post.file_client_ids?.length;`: 
The problem with short-circuit checks is that even if `0` as a value is falsy the `&&` will return the first falsy value encountered, so in this case we could end up returning `0` instead of the expected `false`. It only works flawless when combined with `||`.

Reply:
Thank you!
But this causes the following error as we compare Number which may be undefined and 0.
`Object is possibly 'undefined'`

Reply:
Oh, you are right. I made a mistake there
But to prevent that we can just use a Boolean call (I have tested this here: https://www.typescriptlang.org/play?#code/C4TwDgpgBA8gRgKygXigbwLACgq6gSwBMAuKAZ2ACd8A7AcwG5s8oBDSygflIuvoG0AukywBfbNgDGAexoUo0xAEZS8JKkw487SqX4ByYEv0AaKIYBM+wSeZ4ipQxArHbYkVNnzFCC6sQo6Ha4Ono2wQQk5sDOwFZuoh5YMnLACogAzP7qQVq4DtGxGabYiRLJXtIANhAAdFXSdAAU+gDE7R1QMS6mUABC0tUQrDRNPkq1Opz1EPTAABYAlIueckP1jS0dnd1xvQNDI2OIFpMc0zVzSysVazUbzW3brV1F+4M1Rz4ZZ1wzV8sgA)
```
return post.id === post.pending_post_id && Boolean(post.file_client_ids?.length);
```